### PR TITLE
fix: add ref to catalog entries

### LINKED ISSUE #142: A recipe sample application should be able to target a specific tag, branch or commit
Today only the repo URL is specified

### PR DESCRIPTION
**Pull request by lstocchi**
This PR adds the ref property so that we can checkout the specific branch/commit. 
A valid ref can be a SHA1 tag or a branch name

it resolves #142

### CODE REVIEW

**Review by jeffmaury — CHANGES_REQUESTED** [review]
Also the ref field should be optional

File: packages/backend/src/ai.json [hunk 1]
```diff
@@ -20,6 +22,8 @@
       "description" : "Summarizer application",
       "name" : "Summarizer",
       "repository": "https://github.com/redhat-et/locallm",
+      "branch": "main",
```
[the remark this answers is not in the document]
Comment (lstocchi) [reply packages/backend/src/ai.json:25]: By reading the definition it says "A valid ref can be a SHA1 tag or a branch name", so I think they just do a simple clone. I don't see how i can checkout if i want to take branch 'prod' until commit 'blabla'.

[the remark this answers is not in the document]
Comment (lstocchi) [reply packages/backend/src/ai.json]: i was sure you needed to clone using the branch. Fixed 👍

**Comment by lstocchi** [comment]
I forgot to make the ref optional. Updating

File: packages/shared/src/models/IRecipe.ts [hunk 2]
```diff
@@ -5,6 +5,7 @@ export interface Recipe {
   description: string;
   icon?: string;
   repository: string;
+  ref: string;
```
Comment (feloy) [inline packages/shared/src/models/IRecipe.ts:8]: I think this should be optional. You could add some test to assert it works with an undefined ref

**Review by jeffmaury — COMMENTED** [review]
Make ref optional otherwise I'm good

**Review by feloy — APPROVED** [review]

**Review by jeffmaury — APPROVED** [review]
LGTM

**Pull request merged**